ironfish/src/wallet/wallet.ts

```diff
@@ -411,6 +411,8 @@ export class Wallet {
 
           assetBalanceDeltas.update(transactionDeltas)
 
+          await this.upsertAssetsFromDecryptedNotes(account, decryptedNotes, tx)
```

Comment:
Should we do this for pending transactions too?

Reply:
Ah yeah, forgot to paste that in. Updated

Reply:
Actually - I think this is ok. We only need to do this for assets available in the chain. Pending mints/burns should have that available before it hits this point. If something isn't confirmed, then we shouldn't need it from the chain.

Reply:
I'm thinking about a pending transaction where one account, _A_, is sending some of a custom asset to another account, _B_, for the first time.

Would _B_ already have that asset in its wallet DB?

Reply:
Fixed! Thanks for pointing this out